src/clr-angular/emphasis/alert/alert.ts

```diff
@@ -3,28 +3,52 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { ChangeDetectorRef, Component, EventEmitter, Input, Optional, Output } from '@angular/core';
+import {
+  ChangeDetectorRef,
+  Inject,
+  ElementRef,
+  Component,
+  EventEmitter,
+  Input,
+  Optional,
+  Output,
+  AfterViewInit,
+  PLATFORM_ID,
+} from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
 
 // providers
 import { AlertIconAndTypesService } from './providers/icon-and-types.service';
 import { MultiAlertService } from './providers/multi-alert.service';
 import { isBooleanAttributeSet } from '../../utils/component/is-boolean-attribute-set';
 import { ClrCommonStringsService } from '../../utils/i18n/common-strings.service';
-
+import { AriaLiveService, AriaLivePoliteness } from '../../utils/a11y/aria-live.service';
 @Component({
   selector: 'clr-alert',
-  providers: [AlertIconAndTypesService],
+  providers: [AlertIconAndTypesService, AriaLiveService],
   templateUrl: './alert.html',
   styles: [':host { display: block; }'],
 })
-export class ClrAlert {
+export class ClrAlert implements AfterViewInit {
   constructor(
     public iconService: AlertIconAndTypesService,
     public cdr: ChangeDetectorRef,
     @Optional() public multiAlertService: MultiAlertService,
-    public commonStrings: ClrCommonStringsService
+    public commonStrings: ClrCommonStringsService,
+    private ariaLiveService: AriaLiveService,
+    @Inject(PLATFORM_ID) private platformId: Object,
+    private el: ElementRef
```

Comment:
Shouldn't we add this to the alert-item component instead? By using the service here, we are going to announce only the content of whatever alert-item is projected

Reply:
I was thinking of the case of `clr-alerts` with multiple `clr-alert`.  Currently, it will announce all of them at initialization no matter they are displayed or not.